internal/command/proxy/proxy.go

```diff
@@ -131,5 +138,43 @@ func run(ctx context.Context) (err error) {
 		params.RemoteHost = fmt.Sprintf("%s.internal", appName)
 	}
 
+	if flag.GetBool(ctx, "watch-stdin") {
+		ctx = watchStdinAndAbortOnClose(ctx)
+	}
+
 	return proxy.Connect(ctx, params)
 }
+
+// Asynchronously watches stdin and abort when it closes.
+//
+// There is no guarantee that a OS process spawning the proxy will
+// terminate it, however the stdin is always closed whtn the parent
+// terminates. This way we make sure there are no zombie processes,
+// especially that they hold onto TCP ports.
+//
+// Note that we don't do this when stdin is TTY, because that prevents
+// the process from being moved to a background job on Unix.
+// See https://github.com/brunch/brunch/issues/998.
+func watchStdinAndAbortOnClose(ctx context.Context) context.Context {
+	ctx, cancel := context.WithCancelCause(ctx)
```

Comment:
nit: no need to derive the context when stdin is not a tty

Reply:
hmm, should it check for stdin tty after all? it is behind a cli flag now.

Reply:
Good call, I don't think we need the check anymore!